var/spack/repos/builtin/packages/python/package.py

```diff
@@ -89,8 +89,19 @@ def python_include_dir(self):
     def site_packages_dir(self):
         return os.path.join(self.python_lib_dir, 'site-packages')
 
-
-    def setup_dependent_environment(self, module, spec, ext_spec):
+    def environment_modifications(self, extension_spec):
+        env = super(Python, self).environment_modifications(extension_spec)
+        if extension_spec is not None:
+            # Set PYTHONPATH to include site-packages dir for the
+            # extension and any other python extensions it depends on.
+            python_paths = []
+            for d in extension_spec.traverse():
+                if d.package.extends(self.spec):
+                    python_paths.append(os.path.join(d.prefix, self.site_packages_dir))
+            env.set_env['PYTHONPATH'] = ':'.join(python_paths)
```

Comment:
Tricky part here is that this is actually setting up the `dependent_spec`'s environment.  So how do you know what goes in `python`'s module and what goes in, e.g., `numpy`'s module?  I think you don't, so we should separate this out like I suggested above.